File: leetcode/2163-kth-distinct-string-in-an-array/solution.js
/**
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
var kthDistinct = function(arr, k) {
    let arrMap = new Map();
    for (let char of arr) {
        arrMap.set(char, (arrMap.get(char) || 0) + 1)
    }
    let count = 0
    for (let char of arr) {
        // console.log(char)
        if (arrMap.get(char) === 1) {
            // console.log(char, count)
            count++
            if (count === k) {
                return char
            }
        }
    }
    return ""
};

File: leetcode/2163-kth-distinct-string-in-an-array/solution.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        arr_map = {}
        for char in arr:
            arr_map[char] = arr_map.get(char, 0) + 1
        count = 0
        for char in arr:
            if arr_map[char] == 1:
                count += 1
                if count == k:
                    return char
        return ""
